isort/output.py

```diff
@@ -149,6 +153,38 @@ def sorted_imports(
                     section_output.append("")  # Empty line for black compatibility
                     section_output.append(section_comment_end)
 
+            if section in config.separate_packages:
```

Comment:
Thanks a lot for this @alex-liang3. I know the complexy of this method was already high, but this pirce of code you added here is a perfect candidate for an extract method. With that we could start the needed refactor here to reduce the complexity. Could you please do this refactor?